packit/api.py

```diff
@@ -2176,6 +2177,12 @@ def run_osh_build(
         else:
             cmd = ["osh-cli", "mock-build", str(srpm_path)]
 
+        if csmock_args is None:
+            csmock_args = self.package_config.csmock_args
+
+        if csmock_args:
+            cmd.append("--csmock-args=" + str(csmock_args))
```

Comment:
How does this play with the spaces and escaping?

```
>>> print("--csmock-args=" + str("a b c"))
--csmock-args=a b c
```

I have a feeling this needs to be wrapped in apostrophes/quotes, but that introduces different problem (apostrophes will take the `args` as is, also there is a chance of user unintentionally closing it…)

Reply:
> I have a feeling this needs to be wrapped in apostrophes/quotes, but that introduces different problem (apostrophes will take the `args` as is, also there is a chance of user unintentionally closing it…)

It should be clearly documented how this value would be interpreted as a CLI argument.

Reply:
On the other, this can be (mis)used to redefine another CLI option..;-)

Reply:
Maybe we can check for white characters but I would rather have it transparent and let it on the user.

Reply:
I think it should be `"--csmock-args=" + shlex.quote(csmock_args))`.  This way the quotes will be added if needed and and inner text escaped accordingly.

Reply:
`shlex.quote()` is called by the `run_command()` function as mentioned in https://github.com/packit/packit/pull/2402#discussion_r1742040218. Are you suggesting we should still call it here?

Reply:
It definitely happens for just string:

https://github.com/packit/packit/blob/6a512158135a18021b7d1a78ed487d3ef9471891/packit/utils/commands.py#L75-L76

Going through the subprocess docs doesn't help much tbf…

Reply:
Ok, I see you already opened a pull request for it: https://github.com/packit/packit/pull/2403

Thanks!